corehq/apps/app_manager/models.py

```diff
@@ -3987,6 +3988,8 @@ class ApplicationBase(VersionedDoc, SnapshotMixin,
     # always false for RemoteApp
     case_sharing = BooleanProperty(default=False)
 
+    language_profiles = DictProperty()
```

Comment:
Oooh, I see. Whew. Ok, yeah I think formalizing the schema with a class is a good call, and would make the code a lot clearer. It will also let you access app.language_profiles[key].name instead of app.language_profiles[key]['name'], which also makes it clearer that there are pre-specified properties (i.e. a schema).